docs/connect/clients/nodejs.md

```diff
@@ -136,166 +136,75 @@ await client.disconnect();
+This event is triggered whenever an error occurs within the client.
 
-```js
-import {AggregateSteps, AggregateGroupByReducers, createClient, SchemaFieldTypes} from 'redis';
-const client = createClient();
-await client.connect();
-```
-
-Create an index.
-
-```js
-try {
-    await client.ft.create('idx:users', {
-        '$.name': {
-            type: SchemaFieldTypes.TEXT,
-            SORTABLE: true
-        },
-        '$.city': {
-            type: SchemaFieldTypes.TEXT,
-            AS: 'city'
-        },
-        '$.age': {
-            type: SchemaFieldTypes.NUMERIC,
-            AS: 'age'
-        }
-    }, {
-        ON: 'JSON',
-        PREFIX: 'user:'
-    });
-} catch (e) {
-    if (e.message === 'Index already exists') {
-        console.log('Index exists already, skipped creation.');
-    } else {
-        // Something went wrong, perhaps RediSearch isn't installed...
-        console.error(e);
-        process.exit(1);
-    }
-}
-```
+**It is crucial to listen for error events.**
 
-Create JSON documents to add to your database.
+If a client does not have at least one error listener registered and an error occurs, that error will be thrown, potentially causing the Node.js process to exit unexpectedly.
+See [the EventEmitter docs](https://nodejs.org/api/events.html#events_error_events) for more details.
 
-```js
-await Promise.all([
-    client.json.set('user:1', '$', {
-        "name": "Paul John",
-        "email": "paul.john@example.com",
-        "age": 42,
-        "city": "London"
-    }),
-    client.json.set('user:2', '$', {
-        "name": "Eden Zamir",
-        "email": "eden.zamir@example.com",
-        "age": 29,
-        "city": "Tel Aviv"
-    }),
-    client.json.set('user:3', '$', {
-        "name": "Paul Zamir",
-        "email": "paul.zamir@example.com",
-        "age": 35,
-        "city": "Tel Aviv"
-    }),
-]);
+```typescript
+const client = createClient({
+  // ... client options
+});
+// Always ensure there's a listener for errors in the client to prevent process crashes due to unhandled errors
+client.on('error', error => {
+    console.error(`Redis client error: ${error}`)
+});
 ```
 
-Let's find user 'Paul` and filter the results by age.
 
-```js
-let result = await client.ft.search(
-    'idx:users',
-    'Paul @age:[30 40]'
-);
-console.log(JSON.stringify(result, null, 2));
-/*
-{
-  "total": 1,
-  "documents": [
-    {
-      "id": "user:3",
-      "value": {
-        "name": "Paul Zamir",
-        "email": "paul.zamir@example.com",
-        "age": 35,
-        "city": "Tel Aviv"
-      }
-    }
-  ]
-}
- */
-```
+#### Handling reconnections
 
-Return only the city field.
+If the socket unexpectedly closes, such as due to network issues, the client rejects all commands already sent, as they might have been executed on the server.
+The rest of pending commands will remain queued in memory until a new socket is established.  

```

Comment:
```suggestion
If network issues or other problems unexpectedly close the socket, the client will reject all commands already sent, since the server might have already executed them.
The rest of the pending commands will remain queued in memory until a new socket is established.  

```